### DISCUSSION TITLE
Integrate Eltako FLGTF55 with EEP A5-09-0C

### DISCUSSION THREAD

**Discussion by heikosommer1** [mak-gitdev/HA_enoceanmqtt#123] [Request] [answered]
Hello!
First of all, I'm very happy with your addon and it is much easier to integrate the enocean devices with it,
 would it be possible to add the Eltako FLGTF55 which uses the profile EEP A5-09-0C?
I tried to add it on my own by enhancing the mapping.yaml and eep.xml. But unfortunately not successfully.
This hardware consists of to sensors this A5-09-0C for TVOC concentration and one sensor for temperature and humidity with profile A5-04-02.
These are the entries in the devices-file.
The device are created (listed in the devices of HA).
[TVOC]
address = 0xFFAA5400
rorg = 0xA5
func = 0x09
type = 0x0C

[TempHum01]
address = 0xFFAA5401
rorg = 0xA5
func = 0x04
type = 0x02
There is no error but also no received data in the enoceanmqtt.log. I have activated the option debug and log_pakets in the configuration. And I also checked the device with dolphin view.

I have attached also the files of mapping and EEP. Hopefully you can help me.
[EEP_and_Mapping.zip](https://github.com/mak-gitdev/HA_enoceanmqtt/files/14074255/EEP_and_Mapping.zip)

This is the link to the Hardware-manual of Eltako:
[https://www.eltako.com/fileadmin/downloads/de/_bedienung/FLGTF55_30055520-2_dt.pdf]

Thanks in advance
Heiko

**Comment by mak-gitdev** [comment]
Hi @heikosommer1, 

Sorry for the delay.

Could you confirm that both EEPs use different addresses ?
I also had a quick look at your mapping and EEP files.
There are missing data.

I'll have a closer look later tonight.

**Reply by heikosommer1** [reply]
No problem regading the delay. Yes the device has one address for the Temp/hum-Sensor and one for the TVOC sensor. I also was surpriced. But I use the device since 2 yeard with FHEM. There I also see 2 devices and the two addresses. And also in dolphin view I see 2 devices/addresses.

**Reply by mak-gitdev** [reply, marked as answer]
Hi @heikosommer1,

Please use the files below as custom EEP and mapping files.
[EEP_and_Mapping.zip](https://github.com/mak-gitdev/HA_enoceanmqtt/files/14165895/EEP_and_Mapping.zip)
It should be ok now, please let me know.

**Reply by heikosommer1** [reply]
Whow, it works. The ppms are displayed. The ugm3 concentration is not available but ins't it only a calculation from ppb to ugm3?

Thank you so much.

**Reply by mak-gitdev** [reply]
Hi @heikosommer1, 

From the A5-09-0C EEP [documentation](http://tools.enocean-alliance.org/EEPViewer/profiles/A5/09/0C/A5-09-0C.pdf), a compliant device sends concentration either in ppb or ug/m3.

From the FLGTF55 documentation you shared, we can see that it sends only VOCT (DB1 = 0x00) and reports the concentration only in ppb (DB0 = 0x0A &rarr; DB0.1 = 0).

The mapping is done so that only one of the ug/m3 and ppb sensors is available depending on the unit reported by the device.
That's why only the ppb sensor is available in the case of the FLGTF55.
